dio/lib/src/dio_mixin.dart

```diff
@@ -602,7 +602,7 @@ abstract class DioMixin implements Dio {
       RequestInterceptorHandler handler,
     ) {
       requestOptions = reqOpt;
-      _dispatchRequest(reqOpt)
```

Comment:
The original code didn't pass the generic T down.

Reply:
Sure, but why would `_dispatchRequest` return a dynamic response?

Reply:
https://github.com/ipcjs/dio/blob/42f644a70e0d000067fa9a378fef906e8c4e1afa/dio/lib/src/dio_mixin.dart#L684

Because `transformer.transformResponse(reqOpt, responseBody)` return a dynamic

Reply:
https://github.com/ipcjs/dio/blob/42f644a70e0d000067fa9a378fef906e8c4e1afa/dio/lib/src/dio_mixin.dart#L606

`handler.resolve(value, true)` also receives a `Response<dynamic>` parameter.

Reply:
The generic type T here is more like a parameter that represents a type and does nothing else.🤔